SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -370,74 +386,211 @@ private SingularityDeployResult getDeployResult(final SingularityRequest request
+                maybeUpdatePendingRequest(pendingDeploy, deploy, request, updatePendingDeployRequest);
+                return new SingularityDeployResult(DeployState.WAITING);
+              } else {
+                // don't check overdue here because we want to give it a chance to enqueue the load
+                // balancer request. the next check will determine its fate.
+                if (configuration.getLoadBalancerUri() == null) {
+                  LOG.warn("Deploy {} required a load balancer URI but it wasn't set", pendingDeploy);
+                  return new SingularityDeployResult(DeployState.FAILED, "No valid load balancer URI was present");
+                }
+
+                Collection<SingularityTaskId> toShutDown = tasksToShutDown(deployProgress, otherActiveTasks, request);
+                final Map<SingularityTaskId, SingularityTask> tasks = taskManager.getTasks(Iterables.concat(deployActiveTasks, toShutDown));
+                final LoadBalancerRequestId lbRequestId = getLoadBalancerRequestId(pendingDeploy);
+
+                updateLoadBalancerStateForTasks(deployActiveTasks, LoadBalancerRequestType.ADD,
+                  new SingularityLoadBalancerUpdate(BaragonRequestState.UNKNOWN, lbRequestId, Optional.<String> absent(), System.currentTimeMillis(), LoadBalancerMethod.PRE_ENQUEUE,
+                    Optional.<String> absent()));
+                SingularityLoadBalancerUpdate enqueueResult = lbClient.enqueue(lbRequestId, request, deploy.get(), getTasks(deployActiveTasks, tasks), getTasks(toShutDown, tasks));
+                updateLoadBalancerStateForTasks(deployActiveTasks, LoadBalancerRequestType.ADD, enqueueResult);
+
+                DeployState deployState = interpretLoadBalancerState(enqueueResult, DeployState.WAITING);
+                if (deployState == DeployState.SUCCEEDED) {
+                  updatePendingDeploy(pendingDeploy, Optional.of(enqueueResult), DeployState.WAITING);
+                  return markStepFinished(pendingDeploy, deploy, otherActiveTasks, request, updatePendingDeployRequest);
+                } else if (deployState == DeployState.WAITING) {
+                  updatePendingDeploy(pendingDeploy, Optional.of(enqueueResult), deployState);
+                  return new SingularityDeployResult(DeployState.WAITING);
+                } else {
+                  return fromLbState(deployState, enqueueResult);
+                }
+              }
+            } else {
+              return markStepFinished(pendingDeploy, deploy, otherActiveTasks, request, updatePendingDeployRequest);
+            }
+          case UNHEALTHY:
+          default:
+            return new SingularityDeployResult(DeployState.FAILED, "Not all tasks for deploy were healthy");
+        }
+      } else {
+        maybeUpdatePendingRequest(pendingDeploy, deploy, request, updatePendingDeployRequest);
+        return checkOverdue(deploy, deployActiveTasks, isDeployOverdue);
       }
```

Comment:
This method is too long now. Refactor code into methods that are named appropriately